airbyte-integrations/connectors/source-snowflake/source_snowflake/streams.py

```diff
@@ -0,0 +1,566 @@
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        latest_record_state = latest_record[self.cursor_field]
+        if self.state is not None and len(self.state) > 0:
+            current_state_value = self.state[self.cursor_field]
+            self._cursor_value = max(latest_record_state, current_state_value) if current_state_value is not None else latest_record_state
+            self.state = {self.cursor_field: self._cursor_value}
+            return {self.cursor_field: self._cursor_value}
+        self._cursor_value = latest_record_state
+        self.state = {self.cursor_field: self._cursor_value}
+        return {self.cursor_field: self._cursor_value}
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    @namespace.setter
+    def namespace(self, namespace):
+        self._namespace = namespace
+
+    @property
+    def name(self):
+        return f"{self.table_object['schema']}.{self.table_object['table']}"
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        """
+            path of request
+        """
+
+        return f"{self.url_base}/{self.url_suffix}"
+
+    @property
+    def url_base(self):
+        return self._url_base
+
+    @property
+    def statement(self):
+        database = self.config["database"]
+        schema = self.table_object["schema"]
+        table = self.table_object["table"]
+
+        return f"SELECT * FROM {database}.{schema}.{table}"
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, cursor_field=None, sync_mode=None, **kwargs) -> Iterable[
+        Optional[Mapping[str, any]]]:
+        if sync_mode == SyncMode.incremental:
+            if isinstance(cursor_field, list) and cursor_field:
+                self.cursor_field = cursor_field[0]
+            elif cursor_field:
+                self.cursor_field = cursor_field
+
+            if stream_state:
+                self._cursor_value = stream_state.get(self.cursor_field)
+
+            yield {self.cursor_field: self._cursor_value}
+        else:
+            yield {}
+
+    def get_updated_statement(self, stream_slice):
+        """
+        Can be used consistently only in request_body_json
+        otherwise we are not sure stream slice is the next slice and _cursor_value is updated with the correct data
+        """
+
+        updated_statement = self.statement
+
+        if stream_slice:
+            # TODO MAKE SURE THE CURSOR IS SINGLE VALUE AND NOT A STARTING AND ENDING VALUE (ex: window)
+            self._cursor_value = stream_slice.get(self.cursor_field, None)
+
+        if self._cursor_value:
```

Comment:
I understand your point, I will consider state when using full refresh. I will add an order by statement to the request when the cursor field is present to avoid data loss.